cli/src/semgrep/commands/ci.py

```diff
@@ -80,38 +77,27 @@ def yield_exclude_paths(requested_patterns: Sequence[str]) -> Iterable[str]:
     yield from patterns
 
 
-@contextmanager
-def fix_head_if_github_action(metadata: GitMeta) -> Iterator[None]:
+def fix_head_if_github_action(metadata: GitMeta) -> None:
     """
     GHA can checkout the incorrect commit for a PR (it will create a fake merge commit),
     so we need to reset the head to the actual PR branch head before continuing.
 
     Assumes cwd is a valid git project and that if we are in github-actions pull_request,
     metadata.head_branch_hash point to head commit of current branch
     """
-    if isinstance(metadata, GithubMeta) and metadata.is_pull_request_event:
-        assert metadata.head_branch_hash is not None  # Not none when github action PR
-        assert metadata.base_branch_hash is not None
+    if not (isinstance(metadata, GithubMeta) and metadata.is_pull_request_event):
+        return
+    assert metadata.head_branch_hash is not None  # Not none when github action PR
+    assert metadata.base_branch_hash is not None
 
-        logger.info("Fixing git state for github action pull request")
+    logger.info("Fixing git state for github action pull request")
 
-        logger.debug("Calling git rev-parse HEAD")
-        stashed_rev = git_check_output(["git", "rev-parse", "HEAD"])
-        logger.debug(f"stashed_rev: {stashed_rev}")
+    logger.debug("Calling git rev-parse HEAD")
+    stashed_rev = git_check_output(["git", "rev-parse", "HEAD"])
+    logger.debug(f"stashed_rev: {stashed_rev}")
 
-        logger.info(
-            f"Not on head ref: {metadata.head_branch_hash}; checking that out now."
-        )
-        git_check_output(["git", "checkout", metadata.head_branch_hash])
-
-        try:
-            yield
-        finally:
-            logger.info(f"Returning to original head revision {stashed_rev}")
-            git_check_output(["git", "checkout", stashed_rev])
```

Comment:
Probably should add to changelog that we no longer revert to original state in GHA